presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1031,6 +1035,58 @@ public void testVirtualBucketing()
         }
     }
 
+    @Test
+    public void testMaterializedViewOptimization()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute("CREATE TABLE orders_partitioned WITH (partitioned_by = ARRAY['ds']) AS " +
+                    "SELECT orderkey, orderpriority, '2020-01-01' as ds FROM orders WHERE orderkey < 1000 " +
+                    "UNION ALL " +
+                    "SELECT orderkey, orderpriority, '2019-01-02' as ds FROM orders WHERE orderkey < 1000");
+
+            assertUpdate("CREATE MATERIALIZED VIEW test_orders_mv WITH (partitioned_by = ARRAY['ds']) AS SELECT orderkey, orderpriority, ds FROM orders_partitioned");
+            assertTrue(getQueryRunner().tableExists(getSession(), "test_orders_mv"));
+            String explainPlan = getExplainPlan("SELECT orderkey from test_orders_mv", LOGICAL);
+            //TODO: Where is mv gone from the logical plan?
+            System.out.println(explainPlan);
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS test_orders_mv");
+            queryRunner.execute("DROP TABLE IF EXISTS orders_partitioned");
+        }
+    }
+
+    @Test
+    public void testMaterializedViewOptimizationWithDerivedFields()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute("CREATE TABLE lineitem_partitioned WITH (partitioned_by = ARRAY['ds']) AS " +
+                    "SELECT discount, extendedprice, '2020-01-01' as ds FROM lineitem WHERE orderkey < 1000 " +
+                    "UNION ALL " +
+                    "SELECT discount, extendedprice, '2020-01-01' as ds FROM lineitem WHERE orderkey < 1000 ");
+
+            assertUpdate("CREATE MATERIALIZED VIEW test_lineitem_mv WITH (partitioned_by = ARRAY['ds']) " +
+                    "AS SELECT SUM(discount*extendedprice) as _discount_multi_extendedprice_, ds FROM lineitem_partitioned " +
+                    "GROUP BY ds");
+
+            assertTrue(getQueryRunner().tableExists(getSession(), "test_lineitem_mv"));
+            String explainPlan = getExplainPlan("SELECT _discount_multi_extendedprice_ from test_lineitem_mv", LOGICAL);
+            //TODO: Where is mv gone from the logical plan?
+            System.out.println(explainPlan);
+
+            //TODO: Fails as the scope has a field with empty name
+            explainPlan = getExplainPlan("SELECT SUM(_discount_multi_extendedprice_) from test_lineitem_mv", LOGICAL);
+            System.out.println(explainPlan);
```

Comment:
The failure is expected. I don't think there is anything wrong for this approach. Note that query `SELECT SUM(_discount_multi_extendedprice_) from test_lineitem_mv` doesn't have an output name. Check my comment below.